ndcube/ndcube.py

```diff
@@ -186,13 +187,27 @@ def __init__(self, data, wcs=None, uncertainty=None, mask=None, meta=None,
 
         # Format extra coords.
         if not extra_coords:
-            extra_coords = ExtraCoords()
+            # Get existing extra_coords if initializing from an NDCube
+            if hasattr(data, "extra_coords"):
+                extra_coords = data.extra_coords
+            else:
+                extra_coords = ExtraCoords()
 
         if not isinstance(extra_coords, ExtraCoords):
             raise TypeError("The extra_coords argument must be a ndcube.ExtraCoords object.")
 
+        # Get existing global_coords if initializing from an NDCube
+        if hasattr(data, "global_coords"):
+            global_coords = data._global_coords
+        else:
+            global_coords = GlobalCoords(self)
+
+        if copy:
+            extra_coords = deepcopy(extra_coords)
+            global_coords = deepcopy(global_coords)
+
         self._extra_coords = extra_coords
-        self._global_coords = GlobalCoords(self)
```

Comment:
If we are allowing the user to provide a separate `ExtraCoords` instance when NOT instantiating from another `NDCube`, should we also allow the user to supply a separate `GlobalCoords` instance via a `global_coords` kwarg?  Or am I forgetting that `GlobalCoords` needs a soft copy of the `NDCube` and not allowing this is a safety feature?  @Cadair?

Reply:
@Cadair: Could you clarify this before I merge?

Reply:
I would say that adding a `global_coords` kwarg is out of scope for this PR, we should probably open an issue and come to a decision there. I am not sure what shape that API should take.